src/DDTrace/Integrations/Lumen/LumenIntegration.php

```diff
@@ -86,8 +86,7 @@ function (SpanData $span, $args) use ($rootSpan, $integration, $appName) {
                         $rootSpan->setTag('lumen.route.name', $routeInfo[1]['as']);
                         $resourceName = $routeInfo[1]['as'];
                     }
-
-                    if (null !== $resourceName) {
+                    if (null !== $resourceName && !\ddtrace_config_url_resource_name_enabled()) {
```

Comment:
Sanity check: it looks like this is done in `\DDTrace\trace_method('Laravel\Lumen\Application', 'handleFoundRoute', ...)`, so even with `ddtrace_config_url_resource_name_enabled() === false` there is a span. However, when following the logic I don't see the resource name getting set on this local span, only the  root span. Did I get that wrong?

Reply:
Good that you asked, it is always better to double check. When this setting is false, then the default mechanism steps in, setting it to the FQMN. It is tested and proved here https://github.com/DataDog/dd-trace-php/pull/1318/files#diff-0459d560d408a062aaa303f1f068acfece07fec8200cb9e10c965333f42fd534R50-R53

Reply:
IMO, the nested span should use `$resourceName` regardless of the `ddtrace_config_url_resource_name_enabled` setting. Currently, it does not appear to use this variable on the local span, only the root span, which seems like loss of info. Right?

Reply:
I agree that the info is not on the nested span and I think that is "by design". The information is not lost, though, as it is stored in tags `lumen.route.action` and `lumen.route.name` of the root span, regardless of the setting https://github.com/DataDog/dd-trace-php/blob/labbati/lumen-resources/src/DDTrace/Integrations/Lumen/LumenIntegration.php#L79-L88

I think that it make sense that such information is stored in the root span because `handleFoundRoute` is just an implementation detail and I think it is there from the times when we were only able to have a hook (and generate a span) or not hooking at all.

Reply:
@morrisonlevi as discussed in PM https://github.com/DataDog/dd-trace-php/pull/1318/commits/9311c13770792dd0b2e2e67f921d51c799f17354